pkg/templates/funcs_test.go

```diff
@@ -57,13 +58,69 @@ func TestTmplInt(t *testing.T) {
 }
 
 func TestTmpObjArray(t *testing.T) {
-	output := TmplObjectArray("b", input)
+	type input struct {
+		path string
+		data interface{}
+	}
+	tt := []struct {
+		name   string
+		input  input
+		output []KeyValuePair
+	}{
+		{
+			name: "nil input",
+			input: input{
+				path: "b",
+				data: nil,
+			},
+			output: []KeyValuePair{},
+		},
+		{
+			name: "empty input",
+			input: input{
+				path: "b",
+				data: fromYaml(""),
+			},
+			output: []KeyValuePair{},
+		},
+		{
+			name: "nested values",
+			input: input{
+				path: "b",
+				data: fromYaml(`
+b:
```

Comment:
Kind of hate this non-indented syntax, but I guess that's the only way to do this in go?

Reply:
Yeah, it's not pretty. I figured that in the tests it was not that important. :) I can move it into a const value in the test, but that moves the input/output expectations away from each other.

Reply:
Yeah it's nice to have it in the tests instead of in const's other places 👍 So yeah let's just use this for now. Ideas on how to make this look nicer are appreciated 😃